src/main.rs

```diff
@@ -160,12 +153,30 @@ fn start_program(sess: &mut Session, elf: &Elf) -> Result<(), anyhow::Error> {
     Ok(())
 }
 
+/// Set rtt to blocking mode
+fn set_rtt_to_blocking(
+    core: &mut Core,
+    main_fn_address: u32,
+    rtt_buffer_address: u32,
+) -> anyhow::Result<()> {
+    core.set_hw_breakpoint(main_fn_address)?;
+    core.run()?;
+    core.wait_for_core_halted(Duration::from_secs(5))?;
+
+    const OFFSET: u32 = 44;
+    const FLAG: u32 = 2; // BLOCK_IF_FULL
```

Comment:
Would it make sense to import `BLOCK_IF_FULL` from `defmt-rtt`? Not that it would change the functionality, but more from a design perspective.

Reply:
`defmt-rtt` will probably break on hosted systems, it targets microcontrollers and exports the RTT and defmt symbols.

Adding a dependency just for a single constant is also not really worth it IMO.

One thing we can do here is rename `FLAG` to `BLOCK_IF_FULL` instead of using a comment for that, that makes this a bit more descriptive.